src/main/java/org/jboss/pnc/cleaner/archiver/BuildArchiver.java

```diff
@@ -0,0 +1,437 @@
+
+        archviedBuildRecord.autoAlign = !alignmentLog.contains(DMANIPULATION_DISABLE_TRUE)
+                && !alignmentLog.contains(DMANIPULATION_DISABLE_TRUE1);
+        archviedBuildRecord.brewPullActive = alignmentLog.contains(DBREW_PULL_ACTIVE_TRUE);
+        if (buildLog.contains(FRONTEND_MAVEN_PLUGIN)) {
+            buildType = "MVN-WRAPPED-NPM";
+        }
+        archviedBuildRecord.buildType = buildType;
+
+        archviedBuildRecord.trimmedBuildLog = buildLog.getTrimmedLog();
+        archviedBuildRecord.trimmedAlignLog = alignmentLog.getTrimmedLog();
+
+        archviedBuildRecord.status = proccessErrors(archviedBuildRecord, build.getStatus(), buildLog, alignmentLog);
+
+        archviedBuildRecord.persist();
+    }
+
+    private ArchviedBuildRecord mapBuild(Build build) throws RemoteResourceException {
+        ArchviedBuildRecord archived = new ArchviedBuildRecord();
+
+        archived.buildRecordId = parseBuildID(build.getId());
+
+        archived.submitTime = build.getSubmitTime();
+        archived.startTime = build.getStartTime();
+        archived.endTime = build.getEndTime();
+        archived.submitYear = build.getSubmitTime().get(ChronoField.YEAR);
+        archived.submitMonth = build.getSubmitTime().get(ChronoField.MONTH_OF_YEAR);
+        archived.submitQuarter = build.getSubmitTime().get(IsoFields.QUARTER_OF_YEAR);
+        archived.lastUpdate = build.getLastUpdateTime();
+
+        archived.status = build.getStatus();
+        archived.temporaryBuild = build.getTemporaryBuild();
+        archived.brewPullActive = build.getBuildConfigRevision().isBrewPullActive();
+        archived.buildcontentID = build.getBuildContentId();
+
+        archived.executionRootName = build.getAttributes().get(Attributes.BUILD_BREW_NAME);
+        archived.executionrootversion = build.getAttributes().get(Attributes.BUILD_BREW_VERSION);
+
+        archived.buildConfigID = parseBuildConfigID(build.getBuildConfigRevision().getId());
+        archived.buildConfigRev = build.getBuildConfigRevision().getRev();
+        archived.buildConfigName = build.getBuildConfigRevision().getName();
+
+        archived.projectID = parseProjectID(build.getProject().getId());
+        archived.projectName = build.getProject().getName();
+
+        archived.userId = parseUserID(build.getUser().getId());
+        archived.username = build.getUser().getUsername();
+
+        archived.buildEnvironmentID = parseEnvironmentID(build.getEnvironment().getId());
+
+        archived.groupBuildID = parseGroubBuildID(getNullableID(build.getGroupBuild()));
+
+        if (build.getProductMilestone() != null) {
+            ProductMilestoneRef productMilestone = build.getProductMilestone();
+            archived.productMilestoneID = parseProductMilestoneID(productMilestone.getId());
+            archived.productMilestoneVersion = productMilestone.getVersion();
+
+            ProductVersion productVersion = getProductVersion(productMilestone.getId());
```

Comment:
Can a build have a null ProductMilestone, but have its buildConfig linked to a ProductVersion? Asking because the original script searched any productVersion (and then product) via the buildConfig. But this might be smarter, not sure!

Reply:
When build is started from a BC, if the BC's is in a Product Version, the current milestone (if any) is assigned to the build.

Product Version is not audited on BC, so it could have changed on the BC since the build run. I don't think we want to use it.